src/replication/replication_control_channel_receiver.cpp

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright (C) 2008 Search Solution Corporation. All rights reserved by Search Solution.
+ *
+ *   This program is free software; you can redistribute it and/or modify
+ *   it under the terms of the GNU General Public License as published by
+ *   the Free Software Foundation; either version 2 of the License, or
+ *   (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program; if not, write to the Free Software
+ *  Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
+ *
+ */
+
+/*
+ * replication_control_channel_receiver.cpp - manages master control channel entries
+ */
+
+#include "replication_control_channel_receiver.hpp"
+
+#include <list>
+#include <mutex>
+
+#include "communication_channel.hpp"
+#include "stream_transfer_sender.hpp"
+#include "thread_manager.hpp"
+#include "thread_daemon.hpp"
+
+namespace cubreplication
+{
+  namespace master_ctrl
```

Comment:
Global variables are instantiated when the program starts, but these resources are only required in the context of a master node.

It would be better to make a class master_ctrl and compose it to master_node. This would create necessary resources when master_node is instantiated.